tests/test_websocket_api.py

```diff
@@ -0,0 +1,87 @@
+from tests.conftest import DEFAULT_API_TOKEN
+import random
+import websockets
+import json
+import pytest
+import asyncio
+import re
+import rlp
+
+LOCALHOST = "127.0.0.1"
+
+
+def url(port):
+    return f"ws://{LOCALHOST}:{port}/api/v2/messages/websocket"
+
+
+def cookie():
+    return f"X-Auth-Token={DEFAULT_API_TOKEN}"
+
+
+@pytest.fixture
+async def ws_connections(setup_7_nodes):
+    ws1 = await websockets.connect(url(setup_7_nodes["1"]["api_port"]), extra_headers=[("Cookie", cookie())])
+    ws2 = await websockets.connect(url(setup_7_nodes["2"]["api_port"]), extra_headers=[("Cookie", cookie())])
+    ws3 = await websockets.connect(url(setup_7_nodes["3"]["api_port"]), extra_headers=[("Cookie", cookie())])
+    ws4 = await websockets.connect(url(setup_7_nodes["4"]["api_port"]), extra_headers=[("Cookie", cookie())])
+    ws5 = await websockets.connect(url(setup_7_nodes["5"]["api_port"]), extra_headers=[("Cookie", cookie())])
+    yield {"1": ws1, "2": ws2, "3": ws3, "4": ws4, "5": ws5}
+    await ws1.close()
+    await ws2.close()
+    await ws3.close()
+    await ws4.close()
+    await ws5.close()
+
+
+@pytest.fixture
+async def ws2(setup_7_nodes):
+    ws = await websockets.connect(url(setup_7_nodes["4"]["api_port"]), extra_headers=[("Cookie", cookie())])
+    yield ws
+    await ws.close()
```

Comment:
This could be updated to be exception safe for the resource management using a context manager:
```suggestion
    with contextlib.closing(await websockets.connect(url(setup_7_nodes["4"]["api_port"]), extra_headers=[("Cookie", cookie())])) as ws:
        yield ws
```

Reply:
If a test fails and a context manager is not used, resources may be leaked!